core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3697,16 +3698,60 @@ class KafkaApis(val requestChannel: RequestChannel,
     CompletableFuture.completedFuture[Unit](())
   }
 
-  // Just a place holder for now.
   def handleGetTelemetrySubscriptionsRequest(request: RequestChannel.Request): Unit = {
-    requestHelper.sendMaybeThrottle(request, request.body[GetTelemetrySubscriptionsRequest].getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
-    CompletableFuture.completedFuture[Unit](())
+    val subscriptionRequest = request.body[GetTelemetrySubscriptionsRequest]
+
+    if (ClientMetricsReceiverPlugin.instance.isEmpty) {
+      info("Received get telemetry client request, no metrics receiver plugin configured or running with ZK")
+      requestHelper.sendResponseMaybeThrottle(request, requestThrottleMs =>
+        subscriptionRequest.getErrorResponse(requestThrottleMs, Errors.UNSUPPORTED_VERSION.exception))
```

Comment:
@AndrewJSchofield Though the KIP-714 says that `APIVersions` response should mention telemetry APIs are not supported if ClientMetrics Exporter Plugin is not configured. But I didn't get a way to do that in APIVersions API as the API seems very generic which works with version number configured. I might be missing something here and would wait for @junrao's suggestion. For now I have added the check in Telemetry API itself.

Reply:
We can wait for @junrao to weigh in, but I think the handling of `APIVersions` should be as written in the KIP. The whole point is that a client can tell from the `APIVersionsResponse` that there is no need to get a telemetry subscription. For a cluster without client telemetry enabled, there are no additional RPCs when the client connects. With this PR, a 3.7 client will make extra RPCs even when client telemetry is not enabled. This should be avoided if at all possible.

Reply:
Yes, I agree that we need to exclude telemetry requests in `APIVersionsResponse` if ClientMetrics Exporter Plugin is not configured. We could do that by passing in the plugin availability flag through `ApiVersionManager.apiVersionResponse`.

Reply:
@junrao Thanks for the suggestion. Just want to clarify, does adding `clientTelemetryEnabled` in `apiVersionResponse` requires another KIP or we can add that in KIP-714 or nothing to be documented?

Reply:
@apoorvmittal10 : ApiVersionResponse is defined to include all new types of requests supported on the server. So, there is no need to have a separate KIP. There is no need to add clientTelemetryEnabled in ApiVersionResponse. If clientTelemetryEnabled is false on the server, we can just exclude telemetry related API keys from ApiVersionResponse.

Reply:
Thanks for the quick reply @junrao, I am looking into this. I thought the suggestion was to have a flag similar to `zkMigrationEnabled`, but it looks like to not emit APIKeys of telemetry if plugin is missing.

Reply:
Done.